### PR TITLE
IC3Base: Use rel_ind_check in propate and find_highest_frame

### PR DESCRIPTION
**Pull request by makaimann**

### CODE REVIEW

**Comment by makaimann** [comment]
Waiting on cluster run for performance comparison.

**Comment by makaimann** [comment]
Screenshot from 2021-01-07 19-39-24

Runs finished. Only one more solve compared to current master but I think this change also simplifies the code.

**Review by ahmed-irfan — APPROVED** [review]
LGTM

File: engines/ic3base.cpp [hunk 1]
```diff
@@ -641,42 +645,35 @@ bool IC3Base::is_blocked(const ProofGoal * pg)
...
+    assert(c.children.size());
+
+    // NOTE: rel_ind_check works on conjunctions
+    //       need to negate
+    if (rel_ind_check(i + 1, ic3formula_negate(c), gen, false)) {
+      // can push to next frame
+      // got unsat-core based generalization
+      assert(gen.term);
+      assert(gen.children.size());
+      constrain_frame(i + 1, ic3formula_negate(gen), false);
```
Comment (ahmed-irfan) [inline engines/ic3base.cpp:668]: If I understand correctly, we are doing generalization now -- we were not doing it before. RIght?

Comment (makaimann) [reply engines/ic3base.cpp:668]: Yeah that's right. Not the expensive minimizing generalization, but just the cheap unsat-core based generalization that happens automatically in `rel_ind_check`

**Pull request merged**